lib/introspection.js

```diff
@@ -71,28 +76,140 @@ const parseSignature = (
+
+  for (let i = lines.length - 2; i >= 0; i--) {
+    if (lines[i].indexOf('// ') !== 0) break;
+    comments.unshift(lines[i].slice(3));
+  }
+  return comments;
+};
+
+// Parse comments related to function
+//   lines - <string[]>
+// Returns: <object>, function signature
+const parseComments = lines => {
+  const sig = {
+    title: lines[0] || '',
+    description: '',
+    parameters: [],
+    comments: []
+  };
+  let index = 1;
+
+  while (index < lines.length && !lines[index].startsWith(' ')) {
+    sig.description += '\n' + lines[index++];
+  }
+
+  while (index < lines.length && lines[index].startsWith(' ')) {
+    let arg = lines[index++].trimEnd();
+    const len = arg.length;
+    arg = arg.trimStart();
+    const offset = len - arg.length;
+    const [name, rest] = common.section(arg, ' - ');
+
+    if (rest !== '') {
+      let [type, comment] = common.section(rest, ', ');
+      const types = type.split(' | ')
+        .map(s => s.slice(1, s.length - 1));
+      if (types.length === 1) {
+        if (!ALL_TYPES.find(s => types[0].startsWith(s))) {
+          comment = rest;
+          type = '';
+        } else {
+          type = types[0];
+        }
+      } else {
+        type = [];
+        console.log(types);
+        types.forEach(s => {
+          if (ALL_TYPES.includes(s)) type.push(s);
+        });
+      }
+      sig.parameters.push({ name, type, comment, offset });
+    } else {
+      const parameter = sig.parameters.pop();
+      parameter.comment += `\n${' '.repeat(offset + 2)}${arg}`;
+      sig.parameters.push(parameter);
+    }
+  }
+
+  for (let i = index; i < lines.length; i++) {
+    if (NAMED_LINES.find(s => lines[i].startsWith(s))) {
+      const [name, comment] = common.section(lines[i], ':');
+      sig.comments.push({ name, comment });
+    } else {
+      const obj = sig.comments.pop();
+      obj.comment += '\n' + lines[i];
+      sig.comments.push(obj);
+    }
+  }
+  return sig;
+};
+
+// Introspect interface
+//   namespace - <object>, hash of interfaces
+// Returns: <object>, hash of hash of record, { method, title, parameters }
+const introspectOld = (namespace, text) => {
   const inventory = {};
   for (const name in namespace) {
     const iface = namespace[name];
     const methods = {};
-    inventory[name] = methods;
     for (const method in iface) {
       const fn = iface[method];
-      const signature = parseSignature(fn);
+      const signature = parseSignature(fn, text);
       methods[method] = Object.assign({
         method: name + '.' + method
       }, signature);
     }
+    inventory[name] = methods;
+  }
+  return inventory;
+};
+
+
+// Introspect interface
+//   namespace - <object>, hash of interfaces
+//   text - <string>, data to parse
+// Returns: <object>, hash of hash of record, { method, title, parameters }
+const introspect = (namespace, text) => {
+  const inventory = {};
+  for (const name in namespace) {
+    const iface = namespace[name];
+    const methods = {};
```

Comment:
Ditto.

Reply:
It looks like `methods` may store info about `Class`-es, in that case it would be better to rename it to something like `entities` or `comments`, though I am open to a better naming proposal.